spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CreateViewAnalysis.scala

```diff
@@ -0,0 +1,272 @@
+import org.apache.spark.sql.catalyst.plans.logical.views.ResolvedV2View
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.connector.catalog.CatalogManager
+import org.apache.spark.sql.connector.catalog.CatalogPlugin
+import org.apache.spark.sql.connector.catalog.Identifier
+import org.apache.spark.sql.connector.catalog.LookupCatalog
+import org.apache.spark.sql.connector.catalog.ViewCatalog
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.errors.QueryCompilationErrors.toSQLId
+import org.apache.spark.sql.execution.CommandExecutionMode
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.MetadataBuilder
+import org.apache.spark.sql.util.SchemaUtils
+
+/**
+ * Resolve views in CREATE VIEW and ALTER VIEW AS plans and convert them to logical plans.
+ */
+case class CreateViewAnalysis(spark: SparkSession)
+  extends Rule[LogicalPlan] with LookupCatalog {
+
+  protected lazy val catalogManager: CatalogManager = spark.sessionState.catalogManager
+
+  import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+
+  private lazy val isTempView =
+    (nameParts: Seq[String]) => catalogManager.v1SessionCatalog.isTempView(nameParts)
+  private lazy val isTemporaryFunction = catalogManager.v1SessionCatalog.isTemporaryFunction _
+
+  def asViewCatalog(plugin: CatalogPlugin): ViewCatalog = plugin match {
+    case viewCatalog: ViewCatalog =>
+      viewCatalog
+    case _ =>
+      throw QueryCompilationErrors.missingCatalogAbilityError(plugin, "views")
+  }
+
+  def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+    case CreateIcebergView(ResolvedIdentifier(catalog, nameParts), userSpecifiedColumns, comment,
+    properties, originalText, child, allowExisting, replace) =>
+      convertCreateView(
+        catalog = asViewCatalog(catalog),
+        viewInfo = ViewInfo(
+          ident = nameParts,
+          userSpecifiedColumns = userSpecifiedColumns,
+          comment = comment,
+          properties = properties,
+          originalText = originalText),
+        child = child,
+        allowExisting = allowExisting,
+        replace = replace)
+
+    case AlterViewAs(ResolvedV2View(catalog, ident, _), originalText, query) =>
+      convertCreateView(
+        catalog = catalog,
+        viewInfo = ViewInfo(
+          ident = ident,
+          userSpecifiedColumns = Seq.empty,
+          comment = None,
+          properties = Map.empty,
+          originalText = Option(originalText)),
+        child = query,
+        allowExisting = false,
+        replace = true)
+  }
+
+  private case class ViewInfo(ident: Identifier,
+                      userSpecifiedColumns: Seq[(String, Option[String])],
+                      comment: Option[String],
+                      properties: Map[String, String],
+                      originalText: Option[String])
```

Comment:
Nit: looks like IntelliJ is doing something weird. Most of the time these are two indents past the start of the method line.

Reply:
indeed it looks like my scala isn't properly configured to adhere to the indentation being used in other places

Reply:
I've fixed the indentation now